include/ensmallen_bits/function/static_checks.hpp

```diff
@@ -387,6 +387,26 @@ inline void CheckArbitraryFunctionTypeAPI()
 #endif
 }
 
+template<typename... FunctionAndMatTypes, std::size_t I = 0U>
+typename std::enable_if<I < sizeof...(FunctionAndMatTypes) - 2U, void>::type
+CheckArbitraryFunctionTypeAPI()
+{
+#ifndef ENS_DISABLE_TYPE_CHECKS
+  constexpr size_t size = sizeof...(FunctionAndMatTypes);
+  using TupleType = typename std::tuple<FunctionAndMatTypes...>;
+  using FunctionType = typename std::tuple_element<I, TupleType>::type;
+  using MatType = typename std::tuple_element<size - 1, TupleType>::type;
+
+  static_assert(CheckEvaluate<FunctionType, MatType, MatType>::value,
+      "One of the provided FunctionType does not have a correct definition of Evaluate(). "
+      "Please check that the corresponding FunctionType fully satisfies the requirements"
```

Comment:
```suggestion
      "Please check that the corresponding FunctionType fully satisfies the requirements "
```

Missing space at the end of the line.